src/sync/arc_cell.rs

```diff
@@ -1,90 +0,0 @@
-use std::marker::PhantomData;
-use std::mem;
-use std::sync::Arc;
-use std::sync::atomic::{AtomicUsize, Ordering};
-
-/// A type providing atomic storage and retrieval of an `Arc<T>`.
-#[derive(Debug)]
-pub struct ArcCell<T>(AtomicUsize, PhantomData<Arc<T>>);
-
-impl<T> Drop for ArcCell<T> {
-    fn drop(&mut self) {
-        self.take();
-    }
-}
-
-impl<T> ArcCell<T> {
```

Comment:
Why do you think the `ArcCell` is broken? It's certainly not lock-free, but it seems to work fine. I think we should keep it as it's in crossbeam's scope of concurrency primitives.